integrationtest/wildfly/src/test/java/org/hibernate/search/test/integration/jms/transaction/TransactionalJmsMasterSlave.java

```diff
@@ -0,0 +1,179 @@
+		RegisteredMember newMember = memberRegistration.getNewMember();
+		assertNull( "A non registered member should have null ID", newMember.getId() );
+
+		newMember.setName( "Davide D'Alto" );
+		newMember.setEmail( "dd@slave1.fake.email" );
+		memberRegistration.register();
+
+		assertNotNull( "A registered member should have an ID", newMember.getId() );
+	}
+
+	@Test
+	@InSequence(2)
+	@OperateOnDeployment("slave-2")
+	public void registerNewMemberOnSlave2() throws Exception {
+		RegisteredMember newMember = memberRegistration.getNewMember();
+		assertNull( "A non registered member should have null ID", newMember.getId() );
+
+		newMember.setName( "Peter O'Tall" );
+		newMember.setEmail( "po@slave2.fake.email" );
+		memberRegistration.register();
+
+		assertNotNull( "A registered member should have an ID", newMember.getId() );
+	}
+
+	@Test
+	@InSequence(3)
+	@OperateOnDeployment("master")
+	public void registerNewMemberOnMaster() throws Exception {
+		RegisteredMember newMember = memberRegistration.getNewMember();
+		assertNull( "A non registered member should have null ID", newMember.getId() );
+
+		newMember.setName( "Richard Mayhew" );
+		newMember.setEmail( "rm@master.fake.email" );
+		memberRegistration.register();
+
+		assertNotNull( "A registered member should have an ID", newMember.getId() );
+	}
+
+	@Test
+	@InSequence(4)
+	@OperateOnDeployment("slave-1")
+	public void searchNewMembersAfterSynchronizationOnSlave1() throws Exception {
+		assertSearchResult( "Davide D'Alto", search( "Davide" ) );
+		assertSearchResult( "Peter O'Tall", search( "Peter" ) );
+		assertSearchResult( "Richard Mayhew", search( "Richard" ) );
+	}
+
+	@Test
+	@InSequence(5)
+	@OperateOnDeployment("slave-2")
+	public void searchNewMembersAfterSynchronizationOnSlave2() throws Exception {
+		assertSearchResult( "Davide D'Alto", search( "Davide" ) );
+		assertSearchResult( "Peter O'Tall", search( "Peter" ) );
+		assertSearchResult( "Richard Mayhew", search( "Richard" ) );
+	}
+
+	@Test
+	@InSequence(6)
+	@OperateOnDeployment("master")
+	public void searchNewMembersAfterSynchronizationOnMaster() throws Exception {
+		assertSearchResult( "Davide D'Alto", search( "Davide" ) );
+		assertSearchResult( "Peter O'Tall", search( "Peter" ) );
+		assertSearchResult( "Richard Mayhew", search( "Richard" ) );
+	}
+
+	@Test
+	@InSequence(7)
+	@OperateOnDeployment("slave-2")
+	public void rollbackRegisterNewMenberOnSlave2() throws Exception {
+		RegisteredMember newMember = memberRegistration.getNewMember();
+
+		newMember.setName( "Emmanuel Bernard" );
+		newMember.setEmail( "e@slave2.fake.email" );
+		memberRegistration.rollbackedRegister();
+
+		assertNotNull( "A registered member should have an ID", newMember.getId() );
+	}
+
+	@Test
+	@InSequence(8)
+	@OperateOnDeployment("slave-2")
+	public void searchRollbackedMemberAfterSynchronizationOnSlave2() throws Exception {
+		// we need to explicitly wait because we need to detect the *effective* non operation execution (rollback)
```

Comment:
I see that you are only testing this on "slave-2", would it make sense to check from "slave-1" and  "master" as well?